src/content/docs/dev/send-tokens/interchain-tokens/register-existing-token.mdx

```diff
@@ -61,6 +61,71 @@ This function deploys a token connected to your registered token on the home cha
   that all target chains are trustworthy.
 </Callout>
 
+## Execute Code on Token Receipt
+
+When transferring tokens across chains, you might want to execute specific code when the tokens arrive at their destination. This is possible by implementing the `InterchainTokenExecutable` interface in your destination contract.
+
+### Implement the InterchainTokenExecutable Interface
+
+Your destination contract should implement the [InterchainTokenExecutable](https://github.com/axelarnetwork/interchain-token-service/blob/main/contracts/executable/InterchainTokenExecutable.sol) interface:
+
+```solidity
+interface InterchainTokenExecutable {
+    function executeWithInterchainToken(
+        bytes32 commandId,
+        string calldata sourceChain,
+        bytes calldata sourceAddress,
+        bytes calldata data,
+        bytes32 tokenId,
+        address token,
+        uint256 amount
+    ) external;
+}
+```
+
+When tokens are sent to this contract, the `executeWithInterchainToken` function will be automatically called with:
+
+- `amount`: The quantity of tokens transferred
+- `token`: The address of the token contract
+- `data`: Any additional data you included with the transfer
+
+### Send Tokens with Custom Data
+
+When calling `interchainTransfer`, include your custom data in the `metadata` parameter using this format:
+
+```solidity
+bytes metadata = bytes.concat(bytes4(0), yourCustomPayload);
+```
+
+Example usage:
+
+```solidity
+// Encode function parameters or any data you want to pass
+bytes memory yourCustomPayload = abi.encode(param1, param2, param3);
+
+// Format metadata properly for execution
+bytes memory metadata = bytes.concat(bytes4(0), yourCustomPayload);
+
+// Send the interchain transfer with the metadata
+interchainTokenService.interchainTransfer(
+    tokenId,
+    destinationChain,
+    destinationAddress,
+    amount,
+    metadata,
+    gasValue
+);
+```
+
+This will:
+
+1. Transfer tokens to the destination contract
+2. Call `executeWithInterchainToken` on that contract
```

Comment:
create an example for `executeWithInterchainToken()` the way you for `interchainTransfer()`